### ISSUE TITLE
Changes are still applied on a 422 error for optimisticLockKey

### ISSUE THREAD

**Issue by mstokes-f5** [F5Networks/f5-appsvcs-extension#599]
### Environment
 * Application Services Version: 3.34.0
 * BIG-IP Version: 15.1.4.1

### Summary
When there is a hash mismatch while using `optimisticLockKey`, AS3 returns a 422 error. However, the declaration still appears to have been applied.

### Steps To Reproduce
1. Submit the following declaration:
```json
{
    "schemaVersion": "3.14.0",
    "class": "ADC",
    "Tenant": {
        "class": "Tenant",
        "Application": {
            "class": "Application",
            "template": "generic",
            "vip": {
                "class": "Service_HTTP",
                "virtualAddresses": ["192.0.2.0"]
            }
        }
    }
}
```

2. Submit the following declaration (should result in 422 response):
```json
{
    "schemaVersion": "3.14.0",
    "class": "ADC",
    "Tenant": {
        "class": "Tenant",
        "App": {
            "class": "Application",
            "template": "generic",
            "vip": {
                "class": "Service_HTTP",
                "virtualAddresses": ["192.0.2.0"]
            }
        },
        "optimisticLockKey": "foo"
    }
}
```

3. Perform a GET to `/declare` and notice that the application was renamed from `Application` to `App` despite the 422 error.

### Expected Behavior
The declaration should be rejected, and not applied, in the case of an error response such as 422.

### Actual Behavior
The declaration is applied anyways.

**Comment by dstokesf5** [comment]
Thank you for your feedback. I have added this issue to our internal product backlog as AUTOTOOL-3138.

**Comment by dstokesf5** [comment]
This has been resolved in the 3.38.0 release of AS3.
